packages/perseus-core/src/widgets/matcher/matcher-util.ts

```diff
@@ -0,0 +1,61 @@
+import {CoreUtil} from "@khanacademy/perseus-core";
+
+import type {PerseusMatcherWidgetOptions} from "@khanacademy/perseus-core";
+
+type MatcherInfo = {
+    left: ReadonlyArray<string>;
+    right: ReadonlyArray<string>;
+    orderMatters: boolean;
+    problemNum: number | null | undefined;
+};
+
+export const matcherShuffle = (
+    props: MatcherInfo,
+): {left: ReadonlyArray<string>; right: ReadonlyArray<string>} => {
+    // Use the same random() function to shuffle both columns sequentially
+    const rng = CoreUtil.seededRNG(props.problemNum as number);
```

Comment:
Instead of doing an unsafe cast of `problemNum` to `number` here, could we actually provide a default?

```suggestion
    const rng = CoreUtil.seededRNG(props.problemNum ?? 0);
```

`undefined` becomes `0` when converted to an integer using bitwise operations like `<<`, so I'm pretty sure defaulting to 0 preserves behavior.

Reply:
The cast was already there. I'm hesitant to change widget logic during this project.